shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -268,8 +280,11 @@ - (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
     selectedRange.length -= intersectionRange.length;
   }
 
-  [self.text replaceCharactersInRange:replaceRange withString:text];
-  [self setSelectedTextRange:[FlutterTextRange rangeWithNSRange:selectedRange]
+  [self.text
+      replaceCharactersInRange:[FlutterTextRange rangeWithNSRange:replaceRange withText:self.text]
+                                   .range
+                    withString:text];
+  [self setSelectedTextRange:[FlutterTextRange rangeWithNSRange:selectedRange withText:self.text]
```

Comment:
Looks like we pass `nil` for all uses but this one and `setTextInputState`, and we're already doing these checks in `setTextInputState`. Maybe just add the min/max checks here and drop the changes to `FlutterTextRange`?

Reply:
Good idea, moved it out to a method